blueprints/addon/files/ember-cli-build.js

```diff
@@ -4,7 +4,9 @@ const EmberAddon = require('ember-cli/lib/broccoli/ember-addon');
 
 module.exports = function (defaults) {
   const app = new EmberAddon(defaults, {
-    // Add options here
+    <% if (typescript) {%>'ember-cli-babel': { enableTypeScriptTransform: true },
```

Comment:
We also need to set this flag in the addon's `index.js` for it to work on the `.ts` files under `addon/`, right?

Reply:
Oh, right! I had this already in my test addon I played with, but seems I forgot to apply that back to the blueprint 🤦‍♂️. Great catch!

Reply:
We may also need to double check that this works correctly under Embroider. I had a recollection that there were issues with it a while back, and a quick discord search turned up [evidence of that being true back in early 2022](https://github.com/dfreeman/tsc-only/actions/runs/2201587701), but I'm not sure if it's been fixed in the meantime.

Reply:
> I have absolutely no memory of having made that repo or I would have shared it much earlier

No problem, I feel you! 😜
But that's interesting, the changes there seem quite similar to what I have been doing here, so that makes me feel a bit more confident now! 😀

I just tested this with a TS-addon consumed in a TS-Embroider-app (with the new blueprint), and that was working fine actually! Maybe what was failing back then was fixed meanwhile (not sure what that was, given that the GitHub actions log is not available anymore. But I guess when you don't remember the repo, then you also don't remember the error message... :wink:)

Reply:
> not sure what that was, given that the GitHub actions log is not available anymore

I think the issue then was that Embroider wasn't fully honoring the `enableTypeScriptTransform` flag and still required `ember-cli-typescript` to be present in the addon's `dependencies` to process any TS sources correctly.

> Maybe what was failing back then was fixed meanwhile

I think that's likely, since not processing out TS syntax would fail pretty spectacularly as soon as you tried to run the app 😄 
I also just did a quick check with a fresh Embroider app and everything seemed to work correctly, so this bit seems good to go 👍